### ISSUE TITLE
go routine support wait group on previous line

### ISSUE THREAD

**Issue by bmayfi3ld** [bombsimon/wsl#100]
There is often a need to add a value to a wait group and then calling a go routine that will mark done on that wait group as soon as it finishes. Such as

```go
wg.Add(1)
go goRoutineWithDone()

wg.Add(1)
go goRoutineWithDone2()

// waiting for all go routines to finish
wg.Wait()
```

Currently WSL sees this as a problem with needing a whitespace before the go call. I think this should be encouraged. Thoughts?

**Comment by bombsimon** [comment]
I think this is a good idea generally but I want to look at some more patterns. I'm curious that I haven't seen the need for this before given that I use wait groups quote a lot.

Just to clarify, for this to actually be useable I would assume that you would pass the `wg` to the go routine so it can mark it as done? If so, I think the proper solution here would be similar to `for` and `select` and that is that this should be allowed as long as `wg` is passed to the go routine, like this:

```go
wg.Add(1)
go goRoutineWithDone(wg)

// This should also be valid
anyOtherKindOfVar := getVar()
go anyOtherRoutine(anyOtherKindOfVar)
```

WDYT?

**Comment by supernova2468** [comment]
Ah yea. That I messed up the example. I had my wait group attached to a struct and everything is run in methods to not be passing all of these around as parameters all the time. I stripped out the struct for the example but didn't add back in the parameters.  So my code actually looks more like this.

```go
workerPool.wg.Add(1)
go workerPool.doWork()
```

**Comment by bombsimon** [comment]
Gotcha! Yeah I think that would be pretty much the same. With the current design of wsl I think I would've just grabbed assigned or called on previous line and checked for used on current line. Both our examples would pass such test and I think it would be reasonable to allow this.

**Comment by HTechHQ** [comment]
I just have a similar situation and would like to see this change come true :-)

**Comment by bombsimon** [comment]
Added and released in [v3.4.0](https://github.com/bombsimon/wsl/releases/tag/v3.4.0), will automatically be updated next time @dependabot runs in `golangci-lint` repo
